academic_helper/views/schedule.py

```diff
@@ -54,5 +62,8 @@ def post(self, request, *args, **kwargs):
         if "group_choice" in request.POST:
             choice = request.POST["group_choice"]
             return self.on_user_group_choice(choice)
+        if "groups":
+            groups = json.loads(request.POST["groups"])['groups']
+            return self.get_courses_by_groups(groups)
```

Comment:
You dont use it yet, right?
This doesn't need to be an ajax response - this needs to be called while rendering the page (You know that the user will need the data, no need to wait for the request).